### PR TITLE
Add BRRES anti-flicker

### PR DESCRIPTION
**Pull request by riidefi**
Many CTs feature problematic materials that strobe rapidly in ghost races. This patch fixes that.

Example SZS: [here](https://cdn.discordapp.com/attachments/413505687096918047/445341760869367818/build_046_release.szs)
Before/After:
image
image

Example output: 
```
[MSan.c:159] (Verbose) Sanitizing material 0x917764c0
[MSan.c:104] (Verbose) RAS1_IREF references texcoords or texmaps
[MSan.c:171] Correcting GEN_INFO for material16.
[MSan.c:176] Correcting IND_MTX for material16.
[MSan.c:159] (Verbose) Sanitizing material 0x91776b00
[MSan.c:104] (Verbose) RAS1_IREF references texcoords or texmaps
[MSan.c:171] Correcting GEN_INFO for material2.
[MSan.c:176] Correcting IND_MTX for material2.
```

### CODE REVIEW

File: payload/nw4r/g3d/MSan.c [hunk 1]
```diff
@@ -0,0 +1,177 @@
...
+ *   actually send anything.
+ *
+ * - Indirect matrices (control skew) are omitted, leaving
+ *   them undefined.
+ */
+void sanitizeMaterial(uint8_t *pMat) {
+    MSA_VERBOSE_LOG("Sanitizing material %p", pMat);
+
+    /* If the user is confident the material does not flicker, respect that. */
+    if (getMaterialFlag(pMat) & 0x40000000) {
```
Comment (stblr) [inline payload/nw4r/g3d/MSan.c:160]: This is custom, right?

File: payload/nw4r/g3d/MSan.c [hunk 2]
```diff
@@ -0,0 +1,177 @@
...
+enum BPAddressSpace {
+    BPMEM_IND_MTX = 0x06, /* 3x3 parts, column-major */
+    BPMEM_RAS1_SS = 0x25, /* x2 */
+    BPMEM_IREF = 0x27
+};
+
+/* Determine whether or not indirect matrix patching is necessary. */
+static bool isIndPatchNecessary(
+    uint8_t* pData /* Valid pointer to the indirect settings display list (64 bytes) */) {
+    /* Rationle for removal:
```
Comment (stblr) [inline payload/nw4r/g3d/MSan.c:39]: Rationale

File: payload/nw4r/g3d/MSan.c [hunk 3]
```diff
@@ -0,0 +1,177 @@
...
+
+/* Default indirect matrix (skew). An identity matrix is almost always too
+ * intense, so we supply a simple scale matrix where (U, V) is mapped to (U/10,
+ * V/10):
+ *
+ * | .1 .0 .0 |
+ * | .0 .1 .0 |
+ *
+ */
+static const uint8_t __DefaultIndirectMatrix[CMD_SIZE_BP * 3] = {
```
Comment (stblr) [inline payload/nw4r/g3d/MSan.c:66]: It is not recommended to start identifiers with an underscore in C because they are reserved for the implementation.

**Review by stblr — CHANGES_REQUESTED** [review]

File: payload/nw4r/g3d/MSan.c (hunk 1 above)
Comment (riidefi) [reply payload/nw4r/g3d/MSan.c:160]: Yes

File: payload/nw4r/g3d/MSan.c (hunk 3 above)
Comment (riidefi) [reply payload/nw4r/g3d/MSan.c:66]: I'll change it. I think what I was going for (at the time, 2019) was matching N's style, which uses the __ prefix for static data.

File: payload/nw4r/g3d/MSan.c (hunk 2 above)
Comment (riidefi) [reply payload/nw4r/g3d/MSan.c]: Typo, fixed

File: payload/nw4r/g3d/MSan.c (hunk 3 above)
Comment (stblr) [reply payload/nw4r/g3d/MSan.c]: I see

**Review by stblr — APPROVED** [review]

**Pull request merged**